modules/view/main/index.coffee

```diff
@@ -77,7 +77,7 @@ class View
 
       newNodeSet = enterSet.map (d, i) =>
         datum = d.datum
-        element = @new.call @rootSelection, d.datum, i
+        element = @new.call @rootSelection, datum, i, data.indexOf(datum)
```

Comment:
The advanced search needs this 'true index' which is difficult to retrieve outside the view
This new index differs from 'i' because 'i' is the index of the element in the list of elements being added whereas this is the index of the item in the complete dataset.